sdk/core/Azure.Core.TestFramework/README.md

```diff
@@ -402,3 +402,33 @@ To download and unpack all artifacts use the `Download-DevOpsRecordings.ps1` scr
 The `Download-DevOpsRecordings.ps1` would wait for active runs to finish before retrieving artifacts unless `-NoWait` switch is used.
 
 **NOTE:** these scripts require being signed in with Azure CLI (https://docs.microsoft.com/cli/azure/authenticate-azure-cli?view=azure-cli-latest) and access to the internal DevOps project (https://dev.azure.com/azure-sdk/internal/) 
+
+## Note on private/non-virtual fields in your clients (such as _clientDiagnostics) and InternalsVisibleTo
```

Comment:
I think the problem here is not fields it's that `internal virtual Response<SparkSession> GetSparkSession` is internal virtual and is not forwarded to the base implementation.

That's why the fields are null.

Reply:
In theory, you shouldn't be having/testing internal client methods.

Reply:
Hey @pakrym - See [my branch](https://github.com/chamons/azure-sdk-for-net/blob/d8f01a29df29d33dec94b76ccee678b7839c098b/sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkSessionClient.cs#L196) for how I took a first stab at implementing a LRO.

- I wanted to "hide" the versions which returned a session that was not ready and provide versions that returned a LRO to wait on, so I copy pasted the generated code into the customization and made it internal, then provided a public that wrapped.
- That code now in the customization needed to access _clientDiagnostics, which was null in this one test case.

I patched that branch together by looking at what other SDKs _are_ doing, which is not necessarily the same as what I _should_ be doing. 

Advice welcome here.

Reply:
I decided to make SparkSessionOperation/SparkStatementOperation take an ClientDiagnostics argument to remove the need for this property.

Reply:
Yeah, looks like @pakrym is correct. The diagnostic scope interceptor is not doing its thing because this method is not public. If you make it public temporarily, you will see that there are other errors about not using the correct scope name.
If this operation is exposed only via the LRO, do we need the separate scope here or can we just use the LRO scope?

Reply:
Added some questions and comments, thanks!